specification/dnc/resource-manager/Microsoft.DelegatedNetwork/stable/2021-03-15/orchestrators.json

```diff
@@ -0,0 +1,504 @@
+            "type": "string"
+          },
+          "description": "The resource tags."
+        }
+      },
+      "required": [
+        "kind"
+      ],
+      "x-ms-azure-resource": true
+    },
+    "OrchestratorResourceUpdateParameters": {
+      "description": "Parameters for updating a resource.",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The resource tags."
+        }
+      }
+    },
+    "Orchestrators": {
+      "description": "An array of OrchestratorInstance resources.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Orchestrator"
+          },
+          "description": "An array of OrchestratorInstance resources."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of orchestrators.",
+          "readOnly": true
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "Orchestrator": {
+      "description": "Represents an instance of a orchestrator.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/OrchestratorResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/OrchestratorResourceProperties",
+          "description": "Properties of the provision operation request."
+        }
+      }
+    },
+    "OrchestratorResourceProperties": {
+      "description": "Properties of orchestrator",
+      "type": "object",
+      "properties": {
+        "resourceGuid": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource guid."
+        },
+        "provisioningState": {
+          "type": "string",
+          "enum": [
+            "Deleting",
+            "Succeeded",
+            "Failed",
+            "Provisioning"
+          ],
+          "x-ms-enum": {
+            "name": "OrchestratorInstanceState",
+            "modelAsString": true
+          },
+          "readOnly": true,
+          "description": "The current state of orchestratorInstance resource."
+        },
+        "orchestratorAppId": {
+          "description": "AAD ID used with apiserver",
+          "type": "string"
+        },
+        "orchestratorTenantId": {
+          "description": "TenantID of server App ID",
+          "type": "string"
+        },
+        "clusterRootCA": {
+          "description": "RootCA certificate of kubernetes cluster base64 encoded",
+          "type": "string"
+        },
+        "apiServerEndpoint": {
+          "description": "K8s APIServer url. Either one of apiServerEndpoint or privateLinkResourceId can be specified",
+          "type": "string"
+        },
+        "privateLinkResourceId": {
```

Comment:
Incorporating private links typically takes more orchestration than a single property (i.e. requests + approvals).  Can you explain where in the private endpoints/private links scenario this falls according to Networkings docs:  https://microsoft.sharepoint.com/teams/vnet/SitePages/Private-Endpoints.aspx

Reply:
@pilor  The need for private link is customer running their apiserver in private ip and our control plane need a way to access the server from outside. So our RP will create Private endpoint as part of orchestrator resource creation using privatelink resource id. Can directly access private endpoint ip to reach the server. No need for dns integration

 Yes customer will manually approve our request once  orchestrators resource is created.

Reply:
Ok, ensure you are following networking guidelines from that link.  I am not a private link expert so I can't confirm that you are following all the necessary steps